### PR TITLE
Fix clippy lints from 1.66

### PR DESCRIPTION
**Pull request by PoignardAzur**

### CODE REVIEW

**Comment by PoignardAzur** [comment]
Ugh. The PR fails CI test because it mentions a lint that doesn't exist yet. I think the simplest solution is to keep that PR in reserve and retry CI when the next Rust version comes out.

**Comment by jneem** [comment]
Yes, I encountered this in #2290. Also, be careful with the `clamp` lint, because following clippy's advice there can introduce panics.

File: druid/src/scroll_component.rs [hunk 1]
```diff
@@ -244,7 +244,7 @@ impl ScrollComponent {
         let usable_space = viewport_major - major_padding;
 
         let length = (percent_visible * viewport_major).ceil();
-        let length = length.max(bar_min_size).min(usable_space);
+        let length = length.clamp(bar_min_size, usable_space);
```
Comment (xStrom) [inline druid/src/scroll_component.rs:247]: This seems like a potential panic. Usable space could be below the minimum bar size.

**Review by xStrom — COMMENTED** [review]
Rust 1.66 has arrived so we can continue with this now.

This PR does not seem to resolve all clippy issues though, so we can't pass the CI.

File: druid/src/scroll_component.rs (hunk 1 above)
Comment (jneem) [reply druid/src/scroll_component.rs:247]: I complained about this upstream, and it sounded like they were going to change the lint. Dunno if it made it into 1.66 though.

Comment (xStrom) [reply druid/src/scroll_component.rs:247]: The stable clippy still complains about this the same way. It has that *note: clamp will panic if max < min, min.is_nan(), or max.is_nan()* - which is something at least.

Feels like an overeager lint to me. Now that it has reached stable maybe some people will step on this rake and complain.

**Comment by xStrom** [comment]
As this is blocking some other issues I took the initiative to get this done in #2317.

**Pull request closed without merging**